src/test/java/com/scottlogic/AuthorPostSorterTest.java

```diff
@@ -0,0 +1,66 @@
+package com.scottlogic;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Scanner;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class AuthorPostSorterTest {
+
+    AuthorPostSorter aps = new AuthorPostSorter();
+    private UserPost userPost1 = new UserPost("Joe Bloggs",
+            OffsetDateTime.of(2020, 1, 3, 7, 12, 3, 0, ZoneOffset.UTC),
+            "Hello World!", 2);
+
+    private UserPost userPost2 = new UserPost("Joey Bloggs",
+            OffsetDateTime.of(2020, 1, 3, 8, 53, 34, 0, ZoneOffset.UTC),
+            "Another example post.", 1);
+
+    private UserPost userPost3 = new UserPost("Jane Smith",
+            OffsetDateTime.of(2020, 3, 12, 13, 22, 12, 0, ZoneOffset.UTC),
+            "An example of a post \nwith lines breaks.", 3);
+
+    private UserPost userPost4 = new UserPost("May Bloggs",
+            OffsetDateTime.of(2020, 1, 3, 8, 53, 34, 0, ZoneOffset.UTC),
+            "Another example post.", 1);
+
+    private UserPost userPost5 = new UserPost(null,
+            OffsetDateTime.of(2020, 3, 12, 13, 22, 12, 0, ZoneOffset.UTC),
+            "An example of a post \nwith lines breaks.", 3);
```

Comment:
There's a lot of boilerplate code here for creating very similar objects. We're only really bothered about the author, so we can create a private helper method to return a UserPost with a specified author. It would then be better to call this from the individual test methods, so a reader can immediately see what that particular test is doing.

Reply:
Will look at implementing this today